secrets/secrets_test.go

```diff
@@ -79,8 +81,15 @@ func (m *MockConfigMapInterface) Update(configMap *v1.ConfigMap) (*v1.ConfigMap,
 	return nil, nil
 }
 
+func setSecret(secrets *v1.Secret, configVar *model.ConfigurationVariable, value string) {
+	name := util.ConvertNameToKey(configVar.Name)
+	secrets.Data[name] = []byte(value)
+	secrets.Data[name+generatorInputSuffix], _ = json.Marshal(configVar.Generator)
```

Comment:
I don't think we need to worry about it here. You really want to change this to:

```
	var err error
	secrets.Data[name+generatorInputSuffix], err = json.Marshal(configVar.Generator)
	if err != nil {
		panic(err)
	}
```

Given that it is part of the tests, and not the actual program, I think it is fine to assume that `json` is able to marshal simple structures. We would catch it anyways because in several subtests we verify that the result is `NonEmpty`.